Improve docs and messages (#108)
awdeorio/mailmerge: README.md

[![Build Status](https://travis-ci.com/awdeorio/mailmerge.svg?branch=develop)](https://travis-ci.com/awdeorio/mailmerge)
[![codecov](https://codecov.io/gh/awdeorio/mailmerge/branch/develop/graph/badge.svg)](https://codecov.io/gh/awdeorio/mailmerge)

> A simple, command line [mail merge](https://en.wikipedia.org/wiki/Mail_merge) tool.

Reviewer: I think I'd prefer to keep the original text on this one.

Author: Sure, if you say so... I had no idea "mail merge" was a standard term for this, so this looked like a poor choice of words until I googled it (which was after I read all of the README).

Author: Reverted to the original text, assuming that's what you meant with your thumbs up.

> `mailmerge` generates and sends e-mails based on a [jinja2](http://jinja.pocoo.org/docs/latest/templates/) template, filling it with data from a CSV file.

Reviewer: ```suggestion
Write a template email using [jinja2](http://jinja.pocoo.org/docs/latest/templates/) syntax and `mailmerge` will fill it in from a CSV file.
```

Author: I agree this is better. May I suggest mentioning the SMTP functionality explicitly: "[...will fill it in from a CSV file] and, optionally, send it"?

Reviewer: Sure, sounds good

Hi, Myself,

Your number is 17.

> >>> message 1 not sent (dry run)

Reviewer: I see what you're getting at here.  I don't want the user to be confused and think that a message was sent.  I also don't want them to be confused and think there was an error.  How about this?

```suggestion
>>> message 1 sent (dry run)
```

Author: Well, to me _"sent (dry run)"_ sounds self-contradictory and thus more confusing.
"Would be sent, if not for --dry-run" or something to that effect would be OK.
Happy to change to whatever wording you prefer.

Author: Not sure which wording you prefer here; kept the original for now.

Reviewer: Having something that's consistent and parseable by other tools is a plus.  I like how similar `message 1` and `message 1 sent` are.  Let's stick with `>>> message 1 sent (dry run)`
